### PR TITLE
2319 add sidebar to admin decision data page

### PR DESCRIPTION
**Pull request by richpjames**
# Changes in this PR
This PR adds a sidebar to the admin decision data page which contains:

- The status of the decision data (published/draft/submitted)
- The date that the data was last changed
- The user that last changed the data
- A publicly facing link for the data (currently intentionally blank until functionality has been developed)
- A button to edit the data

A button to download the dataset will be added in a subsequent PR

### Before
image

### After
image

### CODE REVIEW

File: src/decisions/admin/decisions.controller.spec.ts [hunk 1]
```diff
@@ -433,11 +433,30 @@ describe('DecisionsController', () => {
         mockTables,
       );
 
+      Object.defineProperties(DecisionDatasetPresenter.prototype, {
+        changedBy: {
```
Comment (richpjames) [inline src/decisions/admin/decisions.controller.spec.ts:437]: This is a workaround for this bug: https://github.com/facebook/jest/issues/9675
TLDR: Jest's spyOn doesn't work for instance getters

Comment (richpjames) [reply src/decisions/admin/decisions.controller.spec.ts:437]: This means we can't assert that the presenter has been called with the expected values as we can't spy on the getters but this functionality is tested, at a higher level, in the e2e tests

File: src/helpers/format-status.helper.spec.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ describe('formatStatus', () => {
     it('returns a yellow tag', () => {
       const i18nService = createMockI18nService();
 
-      const result = formatStatus(OrganisationVersionStatus.Draft, i18nService);
+      const result = formatStatus(DecisionDatasetStatus.Draft, i18nService);
```
Comment (Gweaton) [inline src/helpers/format-status.helper.spec.ts:66]: I'm a bit unclear on the reason behind this change. The commit message suggests this is adding a new status, but weren't we already showing the status in the decision data table? I think we can probably remove this commit.

If you're keen on having a test for this status, it's probably an idea to either do it in its own test, or test a couple of variables in this test with different `Draft` statuses, but I don't think it's necessary here.

File: src/decisions/presenters/decision-dataset.presenter.spec.ts [hunk 3]
```diff
@@ -180,6 +182,33 @@ describe('DecisionDatasetPresenter', () => {
...
+
+        const presenter = new DecisionDatasetPresenter(dataset, i18nService);
+
+        expect(presenter.changedBy).toEqual({
+          name: 'example',
+          email: 'test@test.com',
+        });
+      });
+
+      it("if the dataset doesn't have a user it returns their name and email address", () => {
```
Comment (Gweaton) [inline src/decisions/presenters/decision-dataset.presenter.spec.ts:203]: Should this be `returns null`?

File: cypress/integration/admin/decisions/show.spec.ts [hunk 4]
```diff
@@ -77,5 +79,92 @@ describe('Showing a decision dataset', () => {
...
+            .contains(publicFacingLinkButtonText)
+            .contains('a')
+            .click()
+            .url()
+            .should('contain', '');
+        },
+      );
+    });
+
+    it('I can publish the data by clicking the publish button', () => {
```
Comment (Gweaton) [inline cypress/integration/admin/decisions/show.spec.ts:152]: This is covered by the `publish.spec` now, so I think we can remove it from here.

File: views/admin/decisions/show.njk [hunk 5]
```diff
@@ -26,7 +51,7 @@
               }}
             </li>
           {% endif %}
-          {% if 'submitDecisionData' in permissions and not (datasetStatus === 'submitted') %}
+          {% if 'submitDecisionData' in permissions and not(datasetStatus === 'submitted') %}
```
Comment (Gweaton) [inline views/admin/decisions/show.njk:54]: Was this not working as expected before?

**Review by Gweaton — APPROVED** [review]
This looks good to me - a few minor things but this is good to go after that!

File: src/helpers/format-status.helper.spec.ts (hunk 2 above)
Comment (richpjames) [reply src/helpers/format-status.helper.spec.ts:66]: Yeah looking back I think this test is redundant and removing this change is the best option.

File: views/admin/decisions/show.njk (hunk 5 above)
Comment (richpjames) [reply views/admin/decisions/show.njk:54]: This was formatted automatically on save by Prettier. I opted to go with it as it doesn't seem to make a difference semantically and it's easier to not be fighting Prettier but I am happy to back it out if you'd prefer

File: src/decisions/presenters/decision-dataset.presenter.spec.ts (hunk 3 above)
Comment (richpjames) [reply src/decisions/presenters/decision-dataset.presenter.spec.ts:203]: Yep

File: views/admin/decisions/show.njk (hunk 5 above)
Comment (Gweaton) [reply views/admin/decisions/show.njk:54]: I'm not sure how reliable Prettier is with `njk` files (we've been struggling to find a reliable config), but it might be better to commit things like this in their own commit (e.e. `Format file with prettier`) so it doesn't introduce overhead in another commit. I don't mind this going in now though.

Comment (richpjames) [reply views/admin/decisions/show.njk:54]: Good point - in future I will add formatting in a seperate commit. 
I am going to re-add the space so it's consistent with the rest of the codebase
